_studio/shared/umc/codec/jpeg_dec/src/umc_mjpeg_mfx_decode_base.cpp

```diff
@@ -324,15 +324,15 @@ Status MJPEGVideoDecoderBaseMFX::_GetFrameInfo(const uint8_t* pBitStream, size_t
     int32_t   precision;
     JSS      sampling;
     JCOLOR   color;
-    CMemBuffInput in;
+    std::shared_ptr<CBaseStreamInput> in = std::make_shared<CMemBuffInput>();
     JERRCODE jerr;
 
     if (!m_IsInit)
         return UMC_ERR_NOT_INITIALIZED;
 
-    in.Open(pBitStream,nSize);
+    ((CMemBuffInput*)in.get())->Open(pBitStream,nSize);
 
-    jerr = m_decBase->SetSource(&in);
+    jerr = m_decBase->SetSource(in);
```

Comment:
I believe what you actually want  to do is to move(!) this pointer to another pointer